frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java

```diff
@@ -83,9 +88,85 @@ private void doSlideOut(AnActionEvent anActionEvent, IManagedBranchSnapshot bran
     LOG.debug(() -> "Entering: branchToSlideOut = ${branchToSlideOut}");
     String branchName = branchToSlideOut.getName();
     var project = getProject(anActionEvent);
-    var branchLayoutWriter = getBranchLayoutWriter(anActionEvent);
+
+    LOG.debug("Refreshing repository state");
+    new Task.Backgroundable(project, "Deleting branch if required...") {
+      @Override
+      public void run(ProgressIndicator indicator) {
+        deleteBranchIfRequired(anActionEvent, branchName);
+      }
+    }.queue();
+  }
+
+  @NotUIThreadSafe
+  private void deleteBranchIfRequired(AnActionEvent anActionEvent, String branchName) {
     var gitRepository = getSelectedGitRepository(anActionEvent).getOrNull();
+    var project = getProject(anActionEvent);
+    var slidOutBranchIsCurrent = getCurrentBranchNameIfManaged(anActionEvent)
+        .map(b -> b.equals(branchName))
+        .getOrElse(true);
+
+    if (slidOutBranchIsCurrent) {
+      LOG.debug("Skipping (optional) local branch deletion because it is equal to current branch");
+      getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh();
+      VcsNotifier.getInstance(project)
+          .notifySuccess(
+              format(
+                  getString("action.GitMachete.BaseSlideOutBranchAction.notification.title.slide-out-success.of-current"),
+                  branchName));
+
+    } else if (gitRepository != null) {
+      var root = gitRepository.getRoot();
+      var configValueOption = getDeleteLocalBranchOnSlideOutGitConfigValue(project, root);
+      if (configValueOption.isEmpty()) {
+        GuiUtils.invokeLaterIfNeeded(
+            () -> suggestBranchDeletion(anActionEvent, branchName, gitRepository, project),
+            ModalityState.NON_MODAL);
+      } else if (configValueOption.isDefined()) {
+        var decision = configValueOption.get();
+        handleBranchDeletionDecision(project, branchName, gitRepository, anActionEvent, decision);
+      }
+
+    } else {
+      getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh();
+    }
+  }
+
+  @UIEffect
+  private void suggestBranchDeletion(AnActionEvent anActionEvent, String branchName, GitRepository gitRepository,
+      Project project) {
+    var slideOutOptions = new DeleteBranchOnSlideOutSuggestionDialog(project).showAndGetSlideOutOptions();
+
+    new Task.Backgroundable(project, "Deleting branch if required...") {
+      @Override
+      public void run(ProgressIndicator indicator) {
+        if (slideOutOptions != null) {
+          if (slideOutOptions.shouldDelete()) {
+            slideOutBranch(anActionEvent, branchName);
```

Comment:
But wait, is branch is even slid out when `Keep branch` is pressed? 😯

Reply:
As in, my understanding would be that it should NOT be deleted, but should be slid out 🤔
Maybe even the buttons could say `Slide out & keep branch` and `Slide out & delete branch` (and third option, `Cancel`)... dunno